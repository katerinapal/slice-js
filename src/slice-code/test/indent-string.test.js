import {runAllCombosTests} from './helpers/utils'

/*
test.only(
  'orAndOr(false, false, true, true)',
  require('./helpers/utils').snapSlice(
    require.resolve('./fixtures/logical-expressions'),
    ({orAndOr}) => {
      return [orAndOr(false, false, true, true)]
    },
  ),
)
/* */

runAllCombosTests({
  filename: require.resolve('indent-string'),
  methods: [
    {
      useDefaultExport: true,
      methodName: 'indentString',
      possibleArguments: [
        ['hello world'],
        ['hello world', 2],
        ['hello world', undefined, '@'],
        ['hello world', 4, '@'],
      ],
    },
  ],
})
