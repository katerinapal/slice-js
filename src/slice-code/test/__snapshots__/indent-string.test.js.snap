// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indentString("hello world") && indentString("hello world", , "@") && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") && indentString("hello world", , "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = 1;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") && indentString("hello world", 2) && indentString("hello world", , "@") && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") && indentString("hello world", 2) && indentString("hello world", , "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") && indentString("hello world", 2) && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") && indentString("hello world", 2) 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = ' ';
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = ' ';
	count = 1;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", , "@") && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", , "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	count = 1;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", 2) && indentString("hello world", , "@") && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", 2) && indentString("hello world", , "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;
	count = count === undefined ? 1 : count;

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", 2) && indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = indent === undefined ? ' ' : indent;


	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", 2) 1`] = `
"'use strict';

module.exports = (str, count, indent) => {
	indent = ' ';


	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;

exports[`indentString("hello world", 4, "@") 1`] = `
"'use strict';

module.exports = (str, count, indent) => {

	return str.replace(/^(?!\\\\s*$)/mg, indent.repeat(count));
};"
`;
